### PR TITLE
Fix NPE in InputHandler.parse

### LINKED ISSUE #1567: Exception when disposing the terminal while parsing
Repro:

- Run `yes`
- Kill terminal

VS Code issue: https://github.com/Microsoft/vscode/issues/54132

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #1567

### CODE REVIEW

**Comment by Fonger** [comment]
Could you merge this fix?

**Comment by Tyriar** [comment]
@Fonger it should be merged for 3.6

File: src/Terminal.ts [hunk 1]
```diff
@@ -193,6 +193,7 @@ export class Terminal extends EventEmitter implements ITerminal, IDisposable, II
 
   // Store if user went browsing history in scrollback
   private _userScrolling: boolean;
+  private _isDisposed: boolean = false;
```
Comment (mofux) [inline src/Terminal.ts]: Wouldn't it be better to add this property to the EventEmitter class (the super class of Terminal), so it tracks the disposed status?

**Comment by mofux** [comment]
I've left a comment above, apart from that I'm happy to get this merged.

File: src/Terminal.ts (hunk 1 above)
Comment (Tyriar) [reply src/Terminal.ts]: Great idea :+1:

**Pull request merged**